tests/CsvHelper.Tests/AutoMapping/AutoMappingTests.cs

```diff
@@ -66,13 +66,13 @@ public void ReaderFloatingCultureDeTest()
 				Assert.NotNull(list);
 				Assert.Equal(2, list.Count);
 				var row = list[0];
-				Assert.Equal(1.1f, row.Single, 4);
-				Assert.Equal(2.2d, row.Double, 4);
+				Assert.Equal(1.1f, row.Single, 4f);
```

Comment:
These should take the overload with `int precision` in the third parameter (rather than `float tolerance`)
```suggestion
				Assert.Equal(1.1d, row.Single, 4);
```

Reply:
> I think this one is unaddressed. If taken exactly as suggested (with the `1.1d`) it will choose the correct overload without a compiler error (and them same for the other `row.Single` check on line 74)

line 74 is single that is changed. all other out of scope of this PR

Reply:
let me check

Reply:
sorry, I didn't get your point. Where error exaclty is?

Reply:
With xunit 2.4.1, `Assert.Equal(1.1f, row.Single, 4);` resolves to `Equal(double expected, double actual, int precision)`.

xunit 2.4.2 appears to add `Equal(float expected, float actual, float tolerance)` which makes the `Assert.Equal(1.1f, row.Single, 4);` call ambiguous and gives a compiler error.

We want to keep using the same `Equal(double expected, double actual, int precision)` overload and we do that with `Assert.Equal(1.1d, row.Single, 4);` (i.e change the `1.1f` in the original call to `1.1d`).

In all, we should have:

```csharp
var row = list[0];
Assert.Equal(1.1d, row.Single, 4);
Assert.Equal(2.2d, row.Double, 4);
Assert.Equal(3.3m, row.Decimal);

row = list[1];
Assert.Equal(-0.1d, row.Single, 4);
Assert.Equal(-0.2d, row.Double, 4);
Assert.Equal(-0.3m, row.Decimal);
```

Reply:
sorry, but why "We want to keep using the same overload". I think that copmare double with double and float with float  is the right way to do it

Reply:
Well there is no loss of precision when converting a `float` to a `double` (that's why there is an implicit conversion). The point (sorry for not being clearer) is that `int precision` is the number of decimal places to use when comparing equality (i.e. `4`), and `float tolerance` is the maximum difference allowed between the values for them to be considered equal (i.e. `0.0001`). See:

https://github.com/xunit/assert.xunit/blob/adf018d77ea89003b0c57b9be1ba88d81ef0937e/EqualityAsserts.cs#L121-L131

https://github.com/xunit/assert.xunit/blob/adf018d77ea89003b0c57b9be1ba88d81ef0937e/EqualityAsserts.cs#L215-L225

```csharp
// Any of these is OK
Assert.Equal(1.1d, row.Single, 4); // Equal(double expected, double actual, int precision)
Assert.Equal(1.1f, row.Single, 0.0001f); // Equal(float expected, float actual, float tolerance)
Assert.Equal(1.1f, row.Single, 0.0001); // Equal(double expected, double actual, double tolerance)

// This is not OK
Assert.Equal(1.1f, row.Single, 4f); // Equal(float expected, float actual, float tolerance)
```

Reply:
> Well there is no loss of precision when converting a `float` to a `double` (that's why there is an implicit conversion). The point (sorry for not being clearer) is that `int precision` is the number of decimal places to use when comparing equality (i.e. `4`), and `float tolerance` is the maximum difference allowed between the values for them to be considered equal (i.e. `0.0001`). See:
> 
> https://github.com/xunit/assert.xunit/blob/adf018d77ea89003b0c57b9be1ba88d81ef0937e/EqualityAsserts.cs#L121-L131
> 
> https://github.com/xunit/assert.xunit/blob/adf018d77ea89003b0c57b9be1ba88d81ef0937e/EqualityAsserts.cs#L215-L225
> 
> ```cs
> // Any of these is OK
> Assert.Equal(1.1d, row.Single, 4); // Equal(double expected, double actual, int precision)
> Assert.Equal(1.1f, row.Single, 0.0001f); // Equal(float expected, float actual, float tolerance)
> Assert.Equal(1.1f, row.Single, 0.0001); // Equal(double expected, double actual, double tolerance)
> 
> // This is not OK
> Assert.Equal(1.1f, row.Single, 4f); // Equal(float expected, float actual, float tolerance)
> ```

ok, let switch to double